operator/src/config.rs

```diff
@@ -13,59 +38,65 @@ const SPARK_URL_START: &str = "spark://";
 /// * `node_type` - SparkNodeType (master/worker/history-server)
 /// * `master` - Master SparkNode containing the required settings
 ///
-pub fn adapt_container_command(node_type: &SparkNodeType, master: &SparkNode) -> String {
+pub fn adapt_container_command(node_type: &SparkNodeType, master: &SparkNode) -> Option<String> {
     let mut master_url: String = String::new();
     // only for workers
     if node_type != &SparkNodeType::Worker {
-        return master_url;
+        return None;
     }
     // get all available master selectors
-    'selectors: for selector in &master.selectors {
+    for selector in &master.selectors {
         // check in conf properties and env variables for port
         // conf properties have higher priority than env variables
-        // TODO: add direct spec property for port from crd here
-        // if Some(port) = selector.port { .. continue} else if {...}
-
-        // use config over env
         if let Some(conf) = &selector.config {
-            for config_option in conf {
-                if config_option.name == "spark.master.port" {
-                    master_url.push_str(
-                        format!(
-                            "{}{}:{},",
-                            SPARK_URL_START, selector.node_name, config_option.value
-                        )
-                        .as_str(),
-                    );
-                    // found
-                    continue 'selectors;
-                }
+            if let Some(master) =
```

Comment:
This is a lot of code to handle just one or two options. But I don't have a better way now either so I'm fine with keeping it as it is.

Reply:
I dont see another way, we have multiple options to set the ports, and i need to get the right port to connect the workers properly